container-storage-setup.sh

```diff
@@ -755,26 +756,33 @@ create_partition() {
     Fatal "udevadm settle after partition creation failed. Exiting."
   fi
 
-  if ! wait_for_dev ${dev}1; then
-    Fatal "Partition device ${dev}1 is not available"
+  part=$(query_first_child $dev)
+
+  if ! wait_for_dev ${part}; then
+    Fatal "Partition device ${part} is not available"
   fi
+  echo $part
+}
+
+query_first_child() {
+  lsblk -npl -o NAME "$1" | tail -n +2 | head -1
```

Comment:
Both of them seems exactly the same to me :). `tail -n +2` will return everything except the device and then you take the first partition out of it. e.g.
````
/dev/vdb
/dev/vdb1
/dev/vdb2
/dev/vdb3.
````
`tail` command will return everything except the device (`/dev/vdb`) and then you pick up the first partition using `head` which is `/dev/vdb1`. If your format seems simpler to you, I will change it.